### ISSUE TITLE
email_account.receive email_host = email.split("@")[1] IndexError: list index out of range

### ISSUE THREAD

**Issue by tangymaru** [frappe/erpnext#20833]
## Description of the issue
```
email_account.receive
```
## Context information (for bug reports)
```
Traceback (most recent call last):
  File "/home/seperp/frappe-bench/apps/frappe/frappe/email/doctype/email_account/email_account.py", line 280, in receive
    communication = self.insert_communication(msg, args=args)
  File "/home/seperp/frappe-bench/apps/frappe/frappe/email/doctype/email_account/email_account.py", line 395, in insert_communication
    communication.insert(ignore_permissions=True)
  File "/home/seperp/frappe-bench/apps/frappe/frappe/model/document.py", line 230, in insert
    self.run_before_save_methods()
  File "/home/seperp/frappe-bench/apps/frappe/frappe/model/document.py", line 895, in run_before_save_methods
    self.run_method("validate")
  File "/home/seperp/frappe-bench/apps/frappe/frappe/model/document.py", line 794, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "/home/seperp/frappe-bench/apps/frappe/frappe/model/document.py", line 1065, in composer
    return composed(self, method, *args, **kwargs)
  File "/home/seperp/frappe-bench/apps/frappe/frappe/model/document.py", line 1048, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "/home/seperp/frappe-bench/apps/frappe/frappe/model/document.py", line 788, in <lambda>
    fn = lambda self, *args, **kwargs: getattr(self, method)(*args, **kwargs)
  File "/home/seperp/frappe-bench/apps/frappe/frappe/core/doctype/communication/communication.py", line 65, in validate
    self.set_timeline_links()
  File "/home/seperp/frappe-bench/apps/frappe/frappe/core/doctype/communication/communication.py", line 261, in set_timeline_links
    contacts = get_contacts([self.sender, self.recipients, self.cc, self.bcc])
  File "/home/seperp/frappe-bench/apps/frappe/frappe/core/doctype/communication/communication.py", line 351, in get_contacts
    email = get_email_without_link(email)
  File "/home/seperp/frappe-bench/apps/frappe/frappe/core/doctype/communication/communication.py", line 423, in get_email_without_link
    email_host = email.split("@")[1]
IndexError: list index out of range
```
**Output of `bench version`**
```
erpnext 12.5.2
frappe 12.3.0
```

## Steps to reproduce the issue

1.
2.
3.

### Stacktrace / full error message

```
(paste here)
```

## Additional information

OS version / distribution, `ERPNext` install method, etc.

**Comment by clarkejj** [comment]
This suggests your problem relates to language charset https://discuss.erpnext.com/t/v12-1-email-pull-from-server-not-working-properly/52757/5

**Comment by Sudhegan** [comment]
same error
 list index out of range when trying to send email manually(create new email in communication) or selecting email address not linked to a customer when sending sales invoice.
below is the traceback
install method: manual
OS: Ubuntu 20.04
erpnext: 13.21.0
frappe: 13.21.0
happened since adding email domain and new email account

````
Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 68, in application
    response = frappe.api.handle()
  File "apps/frappe/frappe/api.py", line 55, in handle
    return frappe.handler.handle()
  File "apps/frappe/frappe/handler.py", line 31, in handle
    data = execute_cmd(cmd)
  File "apps/frappe/frappe/handler.py", line 68, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "apps/frappe/frappe/__init__.py", line 1213, in call
    return fn(*args, **newargs)
  File "apps/frappe/frappe/core/doctype/communication/email.py", line 58, in make
    comm = frappe.get_doc({
  File "apps/frappe/frappe/model/document.py", line 240, in insert
    self.run_before_save_methods()
  File "apps/frappe/frappe/model/document.py", line 971, in run_before_save_methods
    self.run_method("validate")
  File "apps/frappe/frappe/model/document.py", line 869, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1161, in composer
    return composed(self, method, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1144, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "apps/frappe/frappe/model/document.py", line 863, in <lambda>
    fn = lambda self, *args, **kwargs: getattr(self, method)(*args, **kwargs)
  File "apps/frappe/frappe/core/doctype/communication/communication.py", line 66, in validate
    self.set_timeline_links()
  File "apps/frappe/frappe/core/doctype/communication/communication.py", line 263, in set_timeline_links
    contacts = get_contacts([self.sender, self.recipients, self.cc, self.bcc], auto_create_contact=create_contact_enabled)
  File "apps/frappe/frappe/core/doctype/communication/communication.py", line 353, in get_contacts
    email = get_email_without_link(email)
  File "apps/frappe/frappe/core/doctype/communication/communication.py", line 425, in get_email_without_link
    email_host = email.split("@")[1]
IndexError: list index out of range
````

**Comment by Sudhegan** [comment]
> This suggests your problem relates to language charset https://discuss.erpnext.com/t/v12-1-email-pull-from-server-not-working-properly/52757/5

this doesn't work as the my.cnf is already configured when installing erpnext as mentioned in the post

**Comment by ankush** [comment]
fixed in FF 13.22

https://github.com/frappe/frappe/pull/15892
